spec/unit/provider/user/windows_adsi_spec.rb

```diff
@@ -60,6 +60,88 @@
     end
   end
 
+  describe "groups_insync?" do
+
+    let(:group1) { stub(:account => 'group1', :domain => '.', :to_s => 'group1sid') }
+    let(:group2) { stub(:account => 'group2', :domain => '.', :to_s => 'group2sid') }
+    let(:group3) { stub(:account => 'group3', :domain => '.', :to_s => 'group3sid') }
+
+    before :each do
+      Puppet::Util::Windows::SID.stubs(:name_to_sid_object).with('group1').returns(group1)
+      Puppet::Util::Windows::SID.stubs(:name_to_sid_object).with('group2').returns(group2)
+      Puppet::Util::Windows::SID.stubs(:name_to_sid_object).with('group3').returns(group3)
+    end
+
+    it "should return false when current is nil" do
+      expect(provider.groups_insync?(nil, ['group2'])).to be_falsey
```

Comment:
style comment, if this fails, rspec will just say true doesn't equal false, which isn't super helpful. You can get better error messages by flipping things around:

```
expect(provider).to be_groups_insync(nil, ['group2'])
```

See https://www.relishapp.com/rspec/rspec-expectations/v/3-2/docs/built-in-matchers/predicate-matchers#matcher-arguments-are-passed-on-to-the-predicate-method